avro/src/main/scala/hydra/avro/convert/SimpleStringToGenericRecord.scala

```diff
@@ -68,10 +77,36 @@ object SimpleStringToGenericRecord {
       case _ => Success(json)
     }
 
-    def toGenericRecordSimple(schema: Schema, useStrictValidation: Boolean): Try[GenericRecord] = {
+    def toGenericRecordSimple(schema: Schema, useStrictValidation: Boolean = false): Try[GenericRecord] = {
       val jsonValidation = if (useStrictValidation) { checkStrictValidation(str, schema) } else Success()
-      val jsonOfPayload = jsonToGenericRecordJson(str.parseJson, schema)
+      val jsonOfPayload: Try[JsValue] = jsonToGenericRecordJson(str.parseJson, schema)
+
       jsonValidation.flatMap(_ => jsonOfPayload.flatMap(_.compactPrint.toGenericRecordPostValidation(schema)))
     }
   }
+
+  private def defaultToJson(field: Schema.Field): Try[(String, JsValue)] = {
+    /* Matches on types from field.defaultVal() which uses the org.apache.avro.util.internal.JacksonUtils toObject function */
+    def defaultToJsonLoop(defaultVal: AnyRef): Try[JsValue] = {
+       defaultVal match {
+         case null | JsonProperties.NULL_VALUE => Success(JsNull)
+         case b: java.lang.Boolean =>
+           Success(JsBoolean(b))
+         case _: java.lang.Integer | _: java.lang.Float | _: java.lang.Long | _: java.lang.Double =>
+           Success(JsNumber(defaultVal.toString))
+         case s: java.lang.String =>
```

Comment:
If I'm not mistaken, this function precedes the moment where avro wraps strings in Utf8. I'm happy to add that match, but if it did that by default, I'd expect my tests to not function correctly. I do have to wrap the result comparisons in Utf8 to get the matches to work, but that happens after the Json is converted to a GenericRecord. Thoughts?

Reply:
I *think* that the reason for this is because when you are creating the schema you are providing a String type as the default instead of a UTF8. I think that is just passes through whatever you give it. The issue is that Avro will pass a UTF8 there when it creates the schema object when e.g. it fetches the schema from schema registry. You can verify this by creating a schema as json and then using their converter to construct the Schema Java object. I know I've run into this a few times in hydra streams, but it is possible I'm overlooking a difference here

Reply:
Okay turns out Utf8 is only used when you are reading from a generic record and not for a schema's defaults. Approving now.